src/js/components/Form/Form.js

```diff
@@ -85,15 +85,21 @@ const Form = forwardRef(
       setValue(prevValue => {
         const nextValue = { ...prevValue };
         nextValue[name] = data;
-
         // re-run any validations, in case the validation
         // is checking across fields
         setErrors(prevErrors => {
           const nextErrors = { ...prevErrors };
           Object.keys(prevErrors).forEach(errName => {
             if (validations.current[errName]) {
               const nextError = validations.current[errName](data, nextValue);
-              updateErrors(nextErrors, errName, nextError);
+              if (!nextError && nextErrors[name]) {
+                updateErrors(nextErrors, name, nextErrors[name]);
+              } else if (nextError) {
+                updateErrors(nextErrors, errName, nextError);
+              }
+            }
+            if (nextValue[name]) {
+              updateErrors(nextErrors, name, undefined);
```

Comment:
Why do we clear any errors just because there is a value? Won't this potentially clear a validation error set at line 98?